src/core/util/tag.h

```diff
@@ -58,7 +58,13 @@ class Tag
 
   Tag(const std::uint8_t* buf)
   {
-    std::memcpy(m_Buf, buf, Size);
+    if (!buf)
+      {  // Zero-initialize Tag if null-buffer supplied
+        std::array<std::uint8_t, Size> zero_buf{};
+        std::memcpy(m_Buf, zero_buf.data(), zero_buf.size());
+      }
+    else
```

Comment:
> This block and condition make no sense. We should never have a nullptr (but if we did then we should throw here). Why did you remove the assertion and std::exception?

I was mistakenly following Google's style guide, but it is wrong when it comes to exceptions for Kovri. I'm reading Bjarne's papers on exception safety, and will re-add the assert + exception.

Still not quite sure how to handle the cases where `IdentHash`es are function parameters, since passing `nullptr` to those functions also calls this conversion constructor, even when they are const references.

Reply:
>I was mistakenly following Google's style guide

Can you point out where?